From 237cb1cbcb67c37c0b9d72ebb8bcc3a7597e884d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Wed, 20 Aug 2014 08:56:54 +0100
Subject: [PATCH 1/2] Resolves: fdo#86449 backport rtf fixes

empty Reference

valgrind + bff

(cherry picked from commit 0a42632a74596cbc781746931bf8f2650994b80f)

empty m_aStates

valgrind + bff

(cherry picked from commit e3247719911f4e9b61ec43ea1c9ce04bcddc4ff8)

Change-Id: Id3c039a46dec5d2d4a4642dfb53d23a76972dde2
3bd526b7ebf0f4fce5d0c7054809e0dc2908e73f
Reviewed-on: https://gerrit.libreoffice.org/12965
Reviewed-by: Miklos Vajna <vmiklos@collabora.co.uk>
Tested-by: Miklos Vajna <vmiklos@collabora.co.uk>
---
 writerfilter/source/rtftok/rtfdocumentimpl.cxx | 1 +
 1 file changed, 1 insertion(+)

diff --git a/writerfilter/source/rtftok/rtfdocumentimpl.cxx b/writerfilter/source/rtftok/rtfdocumentimpl.cxx
index ca34825..965f914 100644
--- a/writerfilter/source/rtftok/rtfdocumentimpl.cxx
+++ b/writerfilter/source/rtftok/rtfdocumentimpl.cxx
@@ -5059,6 +5059,7 @@ int RTFDocumentImpl::popState()
         case DESTINATION_SHAPEPROPERTYNAME:
         case DESTINATION_SHAPEPROPERTYVALUE:
         case DESTINATION_SHAPEPROPERTY:
+            if (!m_aStates.empty())
             {
                 m_aStates.top().aShape = aState.aShape;
                 m_aStates.top().aPicture = aState.aPicture;
-- 
1.9.3

